functions/assert_metadata_os.pp

```diff
@@ -0,0 +1,57 @@
+# Fails a compile if the client system is not compatible with the module's
+# metadata.json
+#
+# @param module_name
+#   The name of the module that should be checked
+#
+# @param match
+#   The level at which to match the release number
+#
+#   * none  -> No match on release (default)
+#   * full  -> Full release must match
+#   * major -> Only the major release must match
+#
+function simplib::assert_metadata_os (
```

Comment:
@jeefberkey `check_metadata_for_os` is the least ambiguous of these but not really better than what's there. `assert` means fail if bad and `metadata_os` isn't very ambiguous.

I started by calling it `validate`, but that turns into something like `validate_os_from_metadata` which I don't really think is any better than what is already there.